homeassistant/components/duckdns/config_flow.py

```diff
@@ -0,0 +1,105 @@
+"""Config flow for DuckDNS integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from . import _update_duckdns
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("domain"): str,
+        vol.Required("access_token"): str,
+    }
+)
+
+
+class PlaceholderHub:
+    """Placeholder class to make tests pass."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+
+    async def authenticate(
+        self, hass: HomeAssistant, domain: str, access_token: str
+    ) -> bool:
+        """Test if we can authenticate with the host."""
+        session = async_get_clientsession(hass)
+        return await _update_duckdns(session, domain, access_token)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    hub = PlaceholderHub()
+
+    if not await hub.authenticate(hass, data["domain"], data["access_token"]):
+        raise InvalidAuth
+
+    # Return info that you want to store in the config entry.
+    return {"title": "DuckDNS"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for duckdns."""
+
+    VERSION = 1
+
+    async def async_step_user(
```

Comment:
Yaml configured duckdns only allowed configuring a single domain for updating, but this implementation doesn't have that limitation.
That's probably fine, but we should not allow setting up the same domain multiple times. Maybe use the domain as unique_id?

Reply:
It seems to me that there is no need to limit the user to only one implementation of DuckDNS, but I added a check so that the same hostname (domain) is not allowed.